bindings/wasm/src/did/wasm_core_verification_method.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::MapStringAny;
+use crate::crypto::WasmKeyType;
+use crate::did::wasm_core_url::WasmCoreDIDUrl;
+use crate::did::WasmCoreDID;
+use crate::did::WasmMethodData;
+use crate::did::WasmMethodType;
+use crate::error::Result;
+use crate::error::WasmResult;
+use identity_iota::crypto::PublicKey;
+use identity_iota::did::VerificationMethod;
+use wasm_bindgen::prelude::*;
+
+/// A DID Document Verification Method.
+#[wasm_bindgen(js_name = CoreVerificationMethod, inspectable)]
+pub struct WasmCoreVerificationMethod(pub(crate) VerificationMethod);
+
+#[wasm_bindgen(js_class = CoreVerificationMethod)]
+impl WasmCoreVerificationMethod {
+  /// Creates a new `CoreVerificationMethod` from the given `did` and public key.
+  #[allow(non_snake_case)]
+  #[wasm_bindgen(constructor)]
+  pub fn new(
+    did: &WasmCoreDID,
+    keyType: WasmKeyType,
+    publicKey: Vec<u8>,
+    fragment: String,
+  ) -> Result<WasmCoreVerificationMethod> {
+    let public_key: PublicKey = PublicKey::from(publicKey);
+    VerificationMethod::new(did.0.clone(), keyType.into(), &public_key, &fragment)
+      .map(Self)
+      .wasm_result()
+  }
+
+  /// Returns a copy of the `CoreDIDUrl` of the `CoreVerificationMethod`'s `id`.
+  #[wasm_bindgen]
+  pub fn id(&self) -> WasmCoreDIDUrl {
+    WasmCoreDIDUrl::from(self.0.id().clone())
+  }
```

Comment:
Still missing `setId` here, it was added to `StardustVerificationMethod` but not `CoreVerificationMethod`.

```suggestion
  /// Sets the id of the `CoreVerificationMethod`.
  #[wasm_bindgen(js_name = setId)]
  pub fn set_id(&mut self, id: &WasmCoreDIDUrl) -> Result<()> {
    self.0.set_id(id.0.clone()).wasm_result()?;
    Ok(())
  }
```